libraries/apollo-compiler/src/main/kotlin/com/apollographql/apollo/compiler/codegen/kotlin/schema/EnumAsSealedInterfaceBuilder.kt

```diff
@@ -91,71 +100,99 @@ internal class EnumAsSealedBuilder(
         .build()
   }
 
-  private fun IrEnum.unknownClassTypeSpec(): TypeSpec {
-    return TypeSpec.classBuilder("UNKNOWN__${simpleName}")
-        .addSuperinterface(unknownValueInterfaceName())
-        .primaryConstructor(unknownValuePrimaryConstructorSpec)
-        .addProperty(unknownValueRawValuePropertySpecWithInitializer)
-        .addModifiers(KModifier.PRIVATE)
+  private fun IrEnum.unknownValueTypeSpec(): TypeSpec {
+    return TypeSpec.classBuilder(UNKNOWN__)
+        .addKdoc("An enum value that isn't known at build time.")
+        .addSuperinterface(selfClassName)
+        .primaryConstructor(
+            FunSpec.constructorBuilder()
+                .addAnnotation(AnnotationSpec.builder(KotlinSymbols.ApolloPrivateEnumConstructor).build())
+                .addParameter(rawValue, KotlinSymbols.String)
+                .build()
+        )
+        .addProperty(
+            PropertySpec.builder(rawValue, KotlinSymbols.String)
+                .addModifiers(KModifier.OVERRIDE)
+                .initializer(rawValue)
+                .build()
+        )
+        .addAnnotation(AnnotationSpec.builder(KotlinSymbols.Suppress).addMember("%S", "ClassName").build())
         .addFunction(
             FunSpec.builder("equals")
                 .addModifiers(KModifier.OVERRIDE)
                 .addParameter(ParameterSpec("other", KotlinSymbols.Any.copy(nullable = true)))
                 .returns(KotlinSymbols.Boolean)
-                .addCode("if (other !is %T) return false\n", unknownValueClassName())
-                .addCode("return this.rawValue == other.rawValue")
+                .addCode("if (other !is $UNKNOWN__) return false\n",)
+                .addCode("return this.$rawValue == other.rawValue")
                 .build()
         )
         .addFunction(
             FunSpec.builder("hashCode")
                 .addModifiers(KModifier.OVERRIDE)
                 .returns(KotlinSymbols.Int)
-                .addCode("return this.rawValue.hashCode()")
+                .addCode("return this.$rawValue.hashCode()")
                 .build()
         )
         .addFunction(
             FunSpec.builder("toString")
                 .addModifiers(KModifier.OVERRIDE)
                 .returns(KotlinSymbols.String)
-                .addCode("return \"UNKNOWN__(${'$'}rawValue)\"")
+                .addCode("return \"$UNKNOWN__(${'$'}$rawValue)\"")
                 .build()
         )
         .build()
   }
 
   private fun IrEnum.safeValueOfFunSpec(): FunSpec {
-    return FunSpec.builder(Identifier.safeValueOf)
+    return FunSpec.builder(safeValueOf)
         .addKdoc(
-            "Returns the [%T] that represents the specified [rawValue].\n" +
-                "Note: unknown values of [rawValue] will return [UNKNOWN__]. You may want to update your schema instead of calling this function directly.\n",
+            """
+            $safeValueOf returns an instance of [%T] representing [$rawValue].
```

Comment:
```suggestion
            Returns an instance of [%T] representing [$rawValue].
```

Nitpick, I'd avoid repeating the fun nam in its doc